src/twisted/web/test/test_flatten.py

```diff
@@ -264,6 +264,32 @@ def test_serializeComment(self) -> None:
         """
         self.assertFlattensImmediately(Comment("foo bar"), b"<!--foo bar-->")
 
+    def test_hasMSOComments(self):
```

Comment:
hmm, I think I got a little carried away with testing - this might be redundant given the test coverage I added to `test_commentEscaping`, but I don't want to keep making modifications after I requested a review (sorry about that). Reviewer, please let me know if you think this test is overkill.